GillianCore/engine/GeneralSemantics/General/GInterpreter.ml

```diff
@@ -127,6 +131,11 @@ struct
            ------------------------------------------------------@]@\n"
           (Sys.time ()) LCmd.pp lcmd State.pp state)
 
+  let is_internal_proc (prog : UP.prog) proc_name =
+    match Prog.get_proc prog.prog proc_name with
+    | Some proc -> proc.proc_internal
+    | None      -> false
```

Comment:
Can this case ever happen ? If you think not, use `failwith` so we can detect inconsistent cases when they happen